### PR TITLE
feat: Update PKCE_REQUIRED to true by default

### PR DESCRIPTION
**Pull request by dopry**
Assuming PKCE is properly implemented this change should be backwards
compatible with clients that do not support PKCE, see:
https://datatracker.ietf.org/doc/html/rfc7636#section-5

applies to  #1122 

## Checklist

- [X] PR only contains one change (considered splitting up PR)
- [x] unit-test added
- [X] documentation updated
- [x] `CHANGELOG.md` updated (only for user relevant changes)
- [X] author name in `AUTHORS`

### CODE REVIEW

**Comment by dopry** [comment]
This is a draft. After reviewing the [compatibility section of RFC7636 Proof Key for Code Exchange by OAuth Public Clients](https://datatracker.ietf.org/doc/html/rfc7636#section-5). It should be backward compatible, so maybe it would be preferable to remove the PKCE setting from userspace and just set it to true in all cases.

**Comment by dopry** [comment]
> This is a draft. After reviewing the [compatibility section of RFC7636 Proof Key for Code Exchange by OAuth Public Clients](https://datatracker.ietf.org/doc/html/rfc7636#section-5). It should be backward compatible, so maybe it would be preferable to remove the PKCE setting from userspace and just set it to true in all cases.

I actually looked at OAuthLib and it looks like this is a flag to require PKCE in a way that isn't officially specification compliant (it lacks backward compatibility).  The questions that remains in my mind are   

1. is the breaking change worth it or should I  just update the documentation to better inform users, maybe add some sort of 'Warning' for users who don't set this to true.
2. If we pursue the  breaking change, should we follow a deprecation cycle like Django and introduce a deprecation warning for a release?

**Comment by n2ygk** [comment]
@dopry Have you run the tox tests locally to see that breaks them? I usually start testing with just a single py and dj version:
```
tox -e py39-dj32
```
That still takes a while but then you can focus on the specific tests that fail and run them in your IDE if necessary to determine what happened. I'm guessing most of the failures are because the test case does not use PKCE.

Regarding it being a breaking change: That's OK for release 2.0 which will have many breaking changes. I don't think it's necessary to put this through a deprecation cycle.

**Comment by n2ygk** [comment]
BTW, If you change the test cases to use PKCE, also make a few failing tests where it's required as well as override the setting to have some successful tests where PKCE_REQUIRED=False.

It looks like test_scopes is the only set of tests that fail.
```
tests/test_scopes.py FFFFFFFF.FF.
```

**Comment by dopry** [comment]
@n2ygk I wasn't asking about the testing. That's easy.  My question is about how to handle the breaking change of defaulting PKCE_REQUIRED to true. Do we need a deprecation cycle, or is it cool to make the change and just document it.

**Comment by n2ygk** [comment]
> @n2ygk I wasn't asking about the testing. That's easy.  My question is about how to handle the breaking change of defaulting PKCE_REQUIRED to true. Do we need a deprecation cycle, or is it cool to make the change and just document it. 
> 
> 

Read all the way to the bottom where I said:

Regarding it being a breaking change: That's OK for release 2.0 which will have many breaking changes. I don't think it's necessary to put this through a deprecation cycle.

**Comment by dopry** [comment]
@n2ygk I missed that paragraph. I believe this should be ready to merge now. I'm assuming that PKCE_REQUIRED as a feature is already well covered by the existing suite of tests. I did add a test to verify that the default value is True.  

Let me know if there is anything you think I missed. 

side note: the test_models is failing for me locally, token clearing doesn't seem to be working properly in my local environment.

**Review by n2ygk — CHANGES_REQUESTED** [review]
Looks good. Just a minor change to CHANGELOG.md please

**Review by n2ygk — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   on using Celery to automate clearing expired tokens.
 
 ### Changed
+* #1129 (**Breaking**) Changed default value of PKCE_REQUIRED to True. This is a **breaking change**. Clients without
+  PKCE enabled will fail to authenticate. This breaks with [section 5 of RFC7636](https://datatracker.ietf.org/doc/html/rfc7636)
+  in favor of the [OAuth2 Security Best Practices for Authorization Code Grants](`Authorization Code Grant <https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics#section-2.1.>`_ )
```
[the remark this answers is not in the document]
Comment (dopry) [reply CHANGELOG.md]: Thanks!

**Pull request merged**